src/v/cloud_topics/level_one/compaction/sink.h

```diff
@@ -35,38 +39,79 @@ class compaction_sink : public compaction::sliding_window_reducer::sink {
 
     ss::future<> finalize() final;
 
+    void set_range_has_tombstones() { _range_has_tombstones = true; }
+
 private:
+    static constexpr size_t max_object_size = 128_MiB;
+
     // Returns `true` if the current object represented by
-    // `_active_staging_file` and `_builder` should be rolled.
+    // `_active_staging_file` and `_builder` should be rolled (i.e the existing
+    // L1 is closed and pushed to `_closed_staging_files` and
+    // `_closed_object_infos`, and a new `_active_staging_file` and `_builder`
+    // are started)
     bool needs_roll() const;
 
-    // Pushes the current object represented by `_active_staging_file` and
-    // `_builder` to the `_committer`. Leaves `_active_staging_file` and
-    // `_builder` as `nullptr`.
-    ss::future<> commit_update_and_roll();
+    ss::future<> initialize_builder();
 
-    // May commit the current object if `needs_roll()`. Leaves
-    // `_active_staging_file` and `_builder` in a set state.
+    // Closes the existing L1 represented by `_active_staging_file` and
+    // `_builder`, pushing it back to `_closed_staging_files_and_md_infos`, and
+    // then reassigning `_active_staging_file` and `_builder` to construct a new
+    // L1 object
+    ss::future<> roll(bool);
+
+    // Calls `roll()` iff `needs_roll() == true`.
     ss::future<> maybe_roll();
 
-    // Resets metadata (base_offset, max_offset, max_timestamp) to uninitalized
-    // values. Must be called after rolling builder/active_staging_file.
-    void reset_metadata();
+    // Returns `true` if the compaction update built up so far in
+    // `_closed_staging_files_and_infos` can be pushed to the
+    // `_committer`. The `_committer` will still decide (based on its
+    // `committing_policy`) when these updates should be finalized and committed
+    // to the `metastore` and cloud storage- this function only decides when the
+    // `sink` has done enough checkpointable compaction work worth pushing to
+    // the `_committer` as an update.
+    bool needs_pushing() const;
```

Comment:
I think these comments are generally focusing too much on low level details, and it's easy to get lost in the details. Can you add a high level comment somewhere that describes pushing, rolling, staging files, and the committer?